source/guides/clear/performance.rst

```diff
@@ -0,0 +1,236 @@
+
+|CL| has become well-known for the performance it can deliver.
+`Phoronix publishes
+Linux performance comparisons <https://www.phoronix.com/scan.php?page=news_topic&q=Clear+Linux>`_
+that include |CL|.
+
+Software build toolchain
+************************
+
+|CL| uses many techniques in its software build toolchain to improve software
+performance, such as aggressive compiler flags and CPU-specific optimizations.
+If maintained manually, these techniques can become complex to support due to
+the volume of packages and the potential for technical drift of package
+performance configurations. The |CL| team built the :ref:`autospec` tool to
+manage this complexity and to apply the techniques used in the software build
+toolchain across the entire project. autospec is available as part of the OS for
+developers to use when they build their own projects on |CL|.
+
+Latest versions of compilers and low-level libraries
+====================================================
+
+|CL| is a rolling release distribution and follows upstream software
+repositories, including compilers and libraries, for updates. |CL| includes
+upstream source-level optimizations as soon as they're available.
+
+A benchmark approach to compiler performance
+============================================
+
+|CL| chooses the compiler used to build each software package on a case-by-case
+basis to maximize performance. Typically, |CL| uses the open source `GNU Compiler
+Collection <https://gcc.gnu.org/>`_ (GCC) which includes the standard low-level
+libraries: `Glibc <https://www.gnu.org/software/libc/>`_ / 
+`libstdc++ <https://gcc.gnu.org/onlinedocs/libstdc++/>`_ for C and C++
+programming languages. If there is a performance advantage, |CL| will build
+packages with `Clang / LLVM <https://clang.llvm.org/>`_.
+
+|CL| uses patched compilers and low-level libraries for exact control of the
+software build. Patches include changes that default to more aggressive
+optimizations or optimizations that haven't yet been merged upstream.
+
+View the full list of patches in the autospec repositories on GitHub:
+
+* https://github.com/clearlinux-pkgs/gcc
+* https://github.com/clearlinux-pkgs/glibc
+* https://github.com/clearlinux-pkgs/llvm
+
+Aggressive compiler flags
+=========================
+
+|CL| uses aggressive
+`compiler flags <https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html>`_ to
+optimize software builds for runtime performance. Some significant flags that
+|CL| often implements are:
+
+`mtune and march <https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html>`_
+	Options used to tune generated code with optimized instructions for specific
+	CPU types instead of creating generic code for maximum compatibility.
+
+	|CL| defines it's minimum hardware requirements to be second-generation
+	Intel® microarchitecture code name Westmere (released in 2010) or later.
+	This enables the OS to build on newer CPU architecture features. Whenever

```

Comment:
Incorrect, this doesn't necessarily enable the OS to build on newer CPU architectures.

It does enable optimizations that are otherwise not possible. i.e. the compiler can do things that work on never architectures but would fail to work on older architectures.

Reply:
Updated in latest commit.